lightning/src/chain/keysinterface.rs

```diff
@@ -571,6 +573,8 @@ impl InMemorySigner {
 		let sighash = hash_to_message!(&bip143::SigHashCache::new(spend_tx).signature_hash(input_idx, &witness_script, descriptor.output.value, SigHashType::All)[..]);
 		let local_delayedsig = secp_ctx.sign(&sighash, &delayed_payment_key);
 
+		if descriptor.output.script_pubkey != witness_script { return Err(()); }
```

Comment:
The `witness_script `is the hash preimage of the segwit output. You can create the correct script with `Address::p2wsh`.

Reply:
Just to be sure, which Network am I supposed to use for the p2wsh method?

Reply:
Network wont matter - the network only changes the address, once we call `script_pubkey()` we've forgotten the network information again.